ui/ui-components/components/record/SampleRecordCard.tsx

```diff
@@ -46,30 +48,64 @@ export interface SampleRecordCardProps {
   reloadKey?: string;
   warning?: string;
   groupId?: string;
+
+  record?: RecordType;
+  groups?: Models.GroupType[];
+  destinations?: Models.DestinationType[];
 }
 
 // This is exported so we can manipulate it in tests, but is unused elsewhere.
 export const isConfigUI = grouparooUiEdition() === "config";
 
+const cookies = new Cookies();
+
 const getCachedSampleRecordId = (modelId: string): string => {
-  return globalThis.localStorage?.getItem(
-    `sampleRecord:${grouparooUiEdition()}:${modelId}`
-  );
+  if (isBrowser()) {
+    const localStorageRecordId = globalThis.localStorage?.getItem(
+      `sampleRecord:${grouparooUiEdition()}:${modelId}`
+    );
+    if (localStorageRecordId) {
+      setCachedSampleRecordId(modelId, localStorageRecordId);
+      globalThis.localStorage?.removeItem(
+        `sampleRecord:${grouparooUiEdition()}:${modelId}`
+      );
+      return localStorageRecordId;
+    } else {
+      return cookies.get(`sampleRecord:${grouparooUiEdition()}`)?.[modelId];
+    }
+  }
+  return undefined;
 };
 
 const setCachedSampleRecordId = (modelId: string, recordId: string): void => {
-  if (recordId) {
-    globalThis.localStorage?.setItem(
-      `sampleRecord:${grouparooUiEdition()}:${modelId}`,
-      recordId
+  if (isBrowser() && recordId) {
+    const existing = cookies.get<Record<string, string>>(
+      `sampleRecord:${grouparooUiEdition()}`
+    );
+    cookies.set(
+      `sampleRecord:${grouparooUiEdition()}`,
+      { ...existing, [modelId]: recordId },
+      {
+        path: "/",
+        sameSite: "strict",
+      }
     );
   }
 };
 
 const clearCachedSampleRecordId = (modelId: string): void => {
-  globalThis.localStorage?.removeItem(
-    `sampleRecord:${grouparooUiEdition()}:${modelId}`
-  );
+  if (isBrowser()) {
+    const existing = cookies.get<Record<string, string>>(
+      `sampleRecord:${grouparooUiEdition()}`
+    );
```

Comment:
Similarly here - I think we want to delete the cookie maybe?  That's the ol' se the expiry to something in the past trick?

Reply:
Related - without setting it explicitly, do these cookies never expire?

Reply:
My recollection is that not setting it might delete it when the browser closed, so we should probably set a time.

Reply:
And same story here. We aren't clearing the entire cookie, just the data associated with a single model.

I will look into how we should manage the cookie exipiration. Agree that having this potentially be cleared on browser close is not ideal.

Reply:
Cookie expiration is now being handled with a `max-age`